source/country.js

```diff
@@ -1,7 +1,6 @@
+Country = Space.domain.ValueObject.extend('Country', {
 
-Country = Space.messaging.Serializable.extend('Country', {
-
-  Constructor: function(country) {
+  Constructor(country) {
 
     // Allow to provide another instance of Country as param
     if(country instanceof Country) {
```

Comment:
Strike my previous comment. It seems that when serializing/deserializing we get instance of Object.

So I would suggest using only this in this instance, it is simple and covers all cases:

`const country = (country && country.code) ? country.code : country;`

And removing this:
`if(country instanceof Country)`